src/3id/index.js

```diff
@@ -10,49 +10,60 @@ const STORAGE_KEY = 'serialized3id_'
 const MUPORT_IPFS = { host: config.muport_ipfs_host, port: config.muport_ipfs_port, protocol: config.muport_ipfs_protocol}
 
 class ThreeId {
-  constructor (serializeState, ethereum, opts) {
+  constructor (serializeState, ethereum, ipfs, opts) {
     this._ethereum = ethereum
+    this._ipfs = ipfs
     this._keyrings = {}
-    opts = Object.assign({ muportIpfs: MUPORT_IPFS }, opts )
     this._init3id(serializeState, opts)
     localstorage.set(STORAGE_KEY + this.managementAddress, this.serializeState())
   }
 
   async signJWT (payload) {
     const settings = {
       signer: this._mainKeyring.getJWTSigner(),
-      //issuer: this.getDid()
-      issuer: this._muport.getDid()
+      issuer: this.getDid()
     }
     return didJWT.createJWT(payload, settings)
   }
 
   getDid () {
-    return this._muport.getDid()
+    return this._muportDID
   }
 
   serializeState () {
     let stateObj = {
       managementAddress: this.managementAddress,
       seed: this._mainKeyring.serialize(),
       spaceSeeds: {},
-      muport: this._muport.serializeState()
     }
     Object.keys(this._keyrings).map(name => {
       stateObj.spaceSeeds[name] = this._keyrings[name].serialize()
     })
     return JSON.stringify(stateObj)
   }
 
-  _init3id (serializeState, opts) {
+  _init3id (serializeState) {
     const state = JSON.parse(serializeState)
     this.managementAddress = state.managementAddress
     this._mainKeyring = new Keyring(state.seed)
-    this._muport = new MuPort(state.muport, opts.muportIPFS || {} )
     Object.keys(state.spaceSeeds).map(name => {
       this._keyrings[name] = new Keyring(state.spaceSeeds[name])
     })
-    this.muportFingerprint = utils.sha256Multihash(this._muport.getDid())
+  }
+
+  async _initMuport (muportIpfs) {
+    let keys = this._mainKeyring.getPublicKeys()
+    const doc = createMuportDocument(keys.signingKey, this.managementAddress, keys.asymEncryptionKey)
+    let docHash = (await this._ipfs.files.add(Buffer.from(JSON.stringify(doc))))[0].hash
+    this._muportDID = 'did:muport:' + docHash
+    this.muportFingerprint = utils.sha256Multihash(this._muportDID)
+    const publishToInfura = async () => {
```

Comment:
why does this use both this._ipfs and ipfsMini?

Reply:
oh yes, one is adding locally, and the other publish to infura as made clear by the function name :)

Reply:
yeah, ideally we can remove the infura one in the future :)